File: ios-app/ecommerceapp/ecommerceapp/model/Product.swift
//
//  Product.swift
//  ecommerceapp
//
//  Created by Camila Ferreira  on 28/04/26.
//

import Foundation


struct Product: Identifiable, Decodable, Hashable{
    var id: String
    var title: String
    var description: String
    var category: String
    var quantity: Int
    var price: Double
    var imageUrl: String?
}

File: order-service/src/main/java/br/com/camilaferreiranas/orderservice/infrastructure/feign/dto/Product.java
package br.com.camilaferreiranas.orderservice.infrastructure.feign.dto;

import lombok.Data;
import org.springframework.data.annotation.Id;

import java.math.BigDecimal;


@Data
public class Product {



    private String id;


    private String title;


    private String description;



    private String imageUrl;


    private Category category;


    private Integer quantity;

    private BigDecimal price;

}
